webapp/channels/src/components/channel_notifications_modal/components/notification_section.tsx

```diff
@@ -1,90 +1,91 @@
-         * Member's desktop_threads notification level
-         */
-        memberThreadsNotificationLevel: PropTypes.string,
-
-        /**
-         * Ignore channel-wide mentions @channel, @here and @all
-         */
-        ignoreChannelMentions: PropTypes.string,
-
-        /**
-         * User's global notification level
-         */
-        globalNotificationLevel: PropTypes.string,
-
-        /**
-         * onChange handles update of desktop notification level
-         */
-        onChange: PropTypes.func.isRequired,
-
-        /**
-         * onChangeThreads handles update of desktop_threads notification level
-         */
-        onChangeThreads: PropTypes.func,
-
-        /**
-         * Submit function to save notification level
-         */
-        onSubmit: PropTypes.func.isRequired,
-
-        /**
-         * Update function to to expand or collapse a section
-         */
-        onUpdateSection: PropTypes.func.isRequired,
-
-        /**
-         * Error string from the server
-         */
-        serverError: PropTypes.string,
-    };
-
-    handleOnChange = (e) => {
+type Props = {
+
+    /**
+     * Notification section
+     */
+    section: string;
+
+    /**
+     * Expand if true, else collapse the section
+     */
+    expand: boolean;
+
+    /**
+     * Member's desktop notification level
+     */
+    memberNotificationLevel: string;
+
+    /**
+     * Member's desktop_threads notification level
+     */
+    memberThreadsNotificationLevel?: string;
+
+    /**
+     * Ignore channel-wide mentions @channel, @here and @all
+     */
+    ignoreChannelMentions?: string;
+
+    /**
+     * User's global notification level
+     */
+    globalNotificationLevel?: string;
+
+    /**
+     * onChange handles update of desktop notification level
+     */
+    onChange: (value: string) => void;
+
+    /**
+     * onChangeThreads handles update of desktop_threads notification level
+     */
+    onChangeThreads?: (value: string) => void | any;
+
+    /**
+     * Submit function to save notification level
+     */
+    onSubmit: () => void;
+
+    /**
+     * Update function to expand or collapse a section
+     */
+    onUpdateSection: (value: string) => void;
+
+    /**
+     * Error string from the server
+     */
+    serverError?: string;
+}
+export default class NotificationSection extends React.PureComponent<Props> {
+    constructor(props: Props) {
+        super(props);
+    }
+
+    handleOnChange = (e: ChangeEvent<HTMLInputElement>) => {
         this.props.onChange(e.target.value);
-    };
+    }
 
-    handleOnChangeThreads = (e) => {
+    handleOnChangeThreads = (e: ChangeEvent<HTMLInputElement>) => {
         const value = e.target.checked ? NotificationLevels.ALL : NotificationLevels.MENTION;
-
-        this.props.onChangeThreads(value);
-    };
+        this.props.onChangeThreads?.(value);
+    }
 
     handleExpandSection = () => {
         this.props.onUpdateSection(this.props.section);
-    };
+    }
```

Comment:
Eslint will complain of semicolon here and can you please check other CI failures

Reply:
done. @M-ZubairAhmed.